dart/lib/src/protocol/level.dart

```diff
@@ -3,6 +3,8 @@ import 'package:meta/meta.dart';
 /// Severity of the logged [Event].
 @immutable
 class SeverityLevel {
+  // TODO: make this as an enum
```

Comment:
Dart enum are very limited. Here this class is the easiest way to attach a String name, and use it for serialization. If you use enum you'll need a method with a `switch case`

Reply:
got it, but then I cant do this: https://github.com/getsentry/sentry-java/blob/5ebd49cbfd17eb66175e9247db6d27e4df84a56b/sentry/src/main/java/io/sentry/DiagnosticLogger.java#L37

which is see which level is higher based on the ordinal number

Reply:
we'll need to add an int argument
```dart
int level;

const SeverityLevel._(this.name, this.level);
```

Reply:
yeah sure, if enum doesn't give this for free, it won't be any useful having an enum

Reply:
looks like an enum has an `index` exactly for these cases.
https://www.tutorialspoint.com/dart_programming/dart_programming_enumeration.htm

@bruno-garcia opinions? I like strongly typed things but I didn't play enough with Dart yet.
I'm also fine with the code snippet above

Reply:
yes enum have index and values, but you still will need a switch case for the name.

Reply:
an index it is, thx for the feedback